### PR TITLE
Add proper teardown of VMware client

### PR DESCRIPTION
**Pull request by shubhamsg199**
### Problem Statement
There wasn't any proper teardown of vmwareclient. we see `DEBUG - vCenter keep-alive: 2024-04-14 07:51:45.515419+00:00`  in the logs

### Solution
Add a fixture which handles the proper teardown

### CODE REVIEW

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6547
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_computeresource_vmware.py -k test_positive_vmware_custom_profile_end_to_end --external-logging
Test Result : ========== 2 failed, 10 deselected, 57 warnings in 2043.05s (0:34:03) ==========
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_computeresource_vmware.py -k test_positive_vmware_custom_profile_end_to_end

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6550
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_computeresource_vmware.py -k test_positive_vmware_custom_profile_end_to_end --external-logging
Test Result : ========== 2 passed, 10 deselected, 71 warnings in 8832.15s (2:27:12) ==========
```

**Review by jyejare — APPROVED** [review]

**Review by shweta83 — APPROVED** [review]

File: pytest_fixtures/component/provision_vmware.py [hunk 1]
```diff
@@ -13,6 +14,17 @@ def vmware(request):
     return versions[getattr(request, 'param', 'vmware8')]
 
 
+@pytest.fixture(scope='module')
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/provision_vmware.py:17]: I think this should be func scope only?

File: tests/foreman/ui/test_computeresource_vmware.py [hunk 2]
```diff
@@ -56,20 +56,17 @@ def _get_normalized_size(size):
...
     For "Local-Ironforge" datastore the string looks Like:
 
         "Local-Ironforge (free: 1.66 TB, prov: 2.29 TB, total: 2.72 TB)"
     """
-    system = VMWareSystem(
-        hostname=vmware.hostname,
-        username=settings.vmware.username,
-        password=settings.vmware.password,
-    )
+    data_store_name = settings.vmware.datastore
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_computeresource_vmware.py:67]: why changing this to variable instead of positional argument?

Comment (shubhamsg199) [reply tests/foreman/ui/test_computeresource_vmware.py]: Right, no needed, removed

**Review by synkd — APPROVED** [review]

**Review by jameerpathan111 — APPROVED** [review]

**Review by pondrejk — APPROVED** [review]

**Pull request merged**